modules/library/metadata/src/main/java/org/geotools/math/Plane.java

```diff
@@ -45,6 +44,86 @@
  * @author Howard Freeland, for algorithmic inspiration
  */
 public class Plane implements Cloneable, Serializable {
+    
+    /** Point3d data structure used to define Plane. */
+    public static class Point3d {
+        double x,y,z;
+        public Point3d( double x, double y, double z){
+            this.x=x;
+            this.y=y;
+            this.z=z;
+        }
+        public Point3d( double array[]) throws IllegalArgumentException {
+            if( array == null){
+                throw new NullPointerException("Vector array required");
+            }
+            if( array.length == 2 ){
+                this.x = array[0];
+                this.y = array[1];
+                this.z = Double.NaN;
+            }
+            else if( array.length == 3 ){
+                this.x = array[0];
+                this.y = array[1];
+                this.z = array[2];
+            }
+            else {
+                throw new IllegalArgumentException("Vector array (length " + array.length +") is not expected length of 3");
+            }
+        }
+        public Point3d( Point2D point ){
```

Comment:
If this Point3d class is *only* for defining planes, then constructors which take x,y data and assume Double.NaN for z are likely unnecessary.  

In IntelliJ, this constructor is unused.